services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -631,7 +645,31 @@ def retry_get_parquet_file_and_size(
             raise err
 
 
-def fill_builder_info(builder: DatasetBuilder, hf_token: Optional[str]) -> None:
+class ParquetFileValidator:
+    def __init__(self, max_row_group_size: int, max_validation: int = 5) -> None:
```

Comment:
I don't validate all the parquet files because:
- we don't want to check the biggest row group of all the dataset, but rather just get the order of magnitude of the size. Otherwise we might end up converting a dataset that has 99% good row groups but 1% that is a bit too big, which is overkill.
- it would be faster without validating everything


I set "5" and not "1" just for a small robustness improvement. Some datasets have files of various sizes so you might end up checking a single file that happens to be too small if you check only the first one. This is arbitrary though and not super important. The idea is just to "check the first files".

Reply:
OK, I didn't get that we were doing a loop on the parquet files. Could you add a comment with these details?